pyodide-build/pyodide_build/pywasmcross.py

```diff
@@ -51,25 +51,28 @@
     __name__ = PYWASMCROSS_ARGS.pop("orig__name__")
 
 
+import dataclasses
 import shutil
 import subprocess
-from collections import namedtuple
 from collections.abc import Iterable, Iterator
 from typing import Literal, NoReturn
 
-ReplayArgs = namedtuple(
-    "ReplayArgs",
-    [
-        "pkgname",
-        "cflags",
-        "cxxflags",
-        "ldflags",
-        "target_install_dir",
-        "builddir",
-        "pythoninclude",
-        "exports",
-    ],
-)
+
+@dataclasses.dataclass(eq=False, order=False, kw_only=True)
+class BuildArgs:
```

Comment:
Probably not critical, but we could make this a Pydantic class (inherit from pydantic.BaseModel and remove the decorator), so that types would be enforced at runtime, and then if you specify the type of `target_install_dir` and `host_install_dir`  as `Path` it will auto-convert to that, so we won't have to do it later. The API would be identical otherwise.

Reply:
Yes, that sounds great. Though probably we need to check the runtime overhead. pywasmcross is invoked thousand of times during the compilation so I think we need to be careful when adding extra imports and some heavy logics.